src/tpm2_wrap.c

```diff
@@ -3401,6 +3360,47 @@ int wolfTPM2_NVRead(WOLFTPM2_DEV* dev, TPM_HANDLE authHandle,
     return wolfTPM2_NVReadAuth(dev, &nv, nvIndex, dataBuf, pDataSz, offset);
 }
 
+int wolfTPM2_NVOpen(WOLFTPM2_DEV* dev, WOLFTPM2_NV* nv, word32 nvIndex,
+    const byte* auth, word32 authSz)
+{
+    int rc = TPM_RC_SUCCESS;
+    TPMS_NV_PUBLIC nvPublic;
+
+    if (dev == NULL || nv == NULL || authSz > sizeof(nv->handle.auth.buffer))
+        return BAD_FUNC_ARG;
+
+    /* build the "handle" */
+    nv->handle.hndl = nvIndex;
+    /* auth can also be set already via nv->handle */
+    if (auth != NULL && authSz > 0) {
+        nv->handle.auth.size = authSz;
+        XMEMCPY(nv->handle.auth.buffer, auth, authSz);
+    }
+
+    /* Read the NV Index publicArea to have up to date NV Index Name */
+    rc = wolfTPM2_NVReadPublic(dev, nv->handle.hndl, &nvPublic);
+    if (rc != TPM_RC_SUCCESS) {
+    #ifdef DEBUG_WOLFTPM
+        printf("Failed to open (read) NV\n");
+    #endif
+        return TPM_RC_FAILURE;
```

Comment:
Should we return the rc instead of converting to TPM_RC_FAILURE? May help with debug to know the actual rc.